File: dynamic_form_gen/src/main/java/com/ihsinformatics/dynamicformsgenerator/screens/dialogs/NetworkProgressDialog.java
package com.ihsinformatics.dynamicformsgenerator.screens.dialogs;

import android.app.ProgressDialog;
import android.content.Context;

public class NetworkProgressDialog extends ProgressDialog {

	public NetworkProgressDialog(Context context) {
		super(context);
		setCanceledOnTouchOutside(false);
		setCancelable(false);
		setMessage("Please wait...");
		setProgressStyle(ProgressDialog.STYLE_SPINNER);
	}
	
	
	public void show(String message) {
		setMessage(message);
		super.show();
	}

}

File: app/src/main/java/ihsinformatics/com/hydra_mobile/ui/dialogs/NetworkProgressDialog.kt
package ihsinformatics.com.hydra_mobile.ui.dialogs

import android.app.ProgressDialog
import android.content.Context

@Suppress("DEPRECATION")
class NetworkProgressDialog : ProgressDialog {

    constructor(context: Context) : super(context) {
        setCanceledOnTouchOutside(false)
        setCancelable(false)
        setMessage("Please wait...")
        setProgressStyle(android.app.ProgressDialog.STYLE_SPINNER)
    }


    fun show(message: String) {
        setMessage(message)
        super.show()
    }
}